search.html

```diff
@@ -9,13 +10,13 @@
 
 <script>
   window.store = {
-    {% for post in site.posts %}
-      "{{ post.url | slugify }}": {
+    {% for post in site.data._models.blog %}
+      "{{ post.id | permalink: model_dir: 'blogs' }}": {
```

Comment:
Iterate through the model items instead of site.posts and use the standard permalink filter to generate the URL